packages/agent-toolkit/src/core/tools/platform-api-tools/read-docs-tool.ts

```diff
@@ -0,0 +1,237 @@
+      };
+    }
+  }
+
+  // Helper method to determine if the agent should continue paginating
+  private shouldSuggestPagination(docsCount: number, limit: number, currentPage: number): string {
+    if (docsCount === limit) {
+      return `\n\n🔄 PAGINATION SUGGESTION: You received exactly ${limit} documents, which suggests there may be more. Consider calling this tool again with page: ${currentPage + 1} to get additional documents.`;
+    }
+    return '';
+  }
+
+  // Convert docs content to markdown string with pagination metadata
+  private async enrichDocsWithMarkdown(
+    docs: NonNullable<ReadDocsQuery['docs']>,
+    variables: ReadDocsQueryVariables,
+  ): Promise<ToolOutputType<never>> {
+    type ExportMarkdownFromDocMutationVariables = {
+      docId: string;
+      blockIds?: string[];
+    };
+
+    type ExportMarkdownFromDocMutation = {
+      export_markdown_from_doc: {
+        success: boolean;
+        markdown?: string;
+        error?: string;
+      };
+    };
+
+    const docsInfo = await Promise.all(
+      docs
+        .filter((doc): doc is NonNullable<typeof doc> => doc !== null)
+        .map(async (doc) => {
+          // Get markdown content for this doc
+          let blocksAsMarkdown = '';
+          try {
+            const markdownVariables: ExportMarkdownFromDocMutationVariables = {
+              docId: doc.id,
+            };
+
+            const markdownRes = await this.mondayApi.request<ExportMarkdownFromDocMutation>(
+              exportMarkdownFromDoc,
+              markdownVariables,
+            );
+
+            if (markdownRes.export_markdown_from_doc.success && markdownRes.export_markdown_from_doc.markdown) {
+              blocksAsMarkdown = markdownRes.export_markdown_from_doc.markdown;
+            } else {
+              blocksAsMarkdown = `Error getting markdown: ${markdownRes.export_markdown_from_doc.error || 'Unknown error'}`;
+            }
+          } catch (error) {
+            blocksAsMarkdown = `Error getting markdown: ${error instanceof Error ? error.message : 'Unknown error'}`;
+          }
+
+          return {
+            id: doc.id,
+            object_id: doc.object_id,
+            name: doc.name,
+            doc_kind: doc.doc_kind,
+            created_at: doc.created_at,
+            created_by: doc.created_by?.name || 'Unknown',
+            url: doc.url,
+            relative_url: doc.relative_url,
+            workspace: doc.workspace?.name || 'Unknown',
+            workspace_id: doc.workspace_id,
+            doc_folder_id: doc.doc_folder_id,
+            settings: doc.settings,
+            blocks_as_markdown: blocksAsMarkdown,
+          };
+        }),
+    );
+
+    const currentPage = variables.page || 1;
+    const limit = variables.limit || 25;
+    const docsCount = docsInfo.length;
+    const hasMorePages = docsCount === limit; // If we got exactly the limit, there might be more
```

Comment:
redability wise, id call it gotAllPages. its more correct since you really don't know if there are more pages